src/classes/ExtendedGraph.class.php

```diff
@@ -171,24 +177,60 @@ public function add_literal_triple($s, $p, $o, $lang = null, $dt = null) {
      * @return bool
      */
     private function _add_triple($s, $p, Array $o_info) {
-        if (!isset($this->_index[$s])) {
-            $this->_index[$s] = array();
-            $this->_index[$s][$p] = array( $o_info );
-            return true;
-        }
-        elseif (!isset($this->_index[$s][$p])) {
-            $this->_index[$s][$p] = array( $o_info);
-            return true;
-        }
-        else {
-            if ( ! in_array( $o_info, $this->_index[$s][$p] ) ) {
-                $this->_index[$s][$p][] = $o_info;
+        // Make sure the subject is not an empty string
+        if($s === ""){
```

Comment:
I was just mooting that with Ross in a comment further down. Wondering if I should break up ``isValidTripleValue`` into more specific validation methods for the subject, predicate and value.